modules/api-svc/src/InboundServer/handlers.js

```diff
@@ -937,12 +950,39 @@ const putBulkTransfersByIdError = async(ctx) => {
     ctx.response.body = '';
 };
 
-
 const healthCheck = async(ctx) => {
     ctx.response.status = ReturnCodes.OK.CODE;
     ctx.response.body = '';
 };
 
+const postFxQuotes = async (ctx) => {
+    const sourceFspId = ctx.request.headers['fspiop-source'];
+    const { body, headers } = ctx.request;
+    const { logger } = ctx.state;
+    const logPrefix = 'Handling POST fxQuotes request';
+
+    const model = createInboundTransfersModel(ctx);
+    model.postFxQuotes({ body, headers }, sourceFspId)
+        .then(response => logger.push({ response }).log(`${logPrefix} is done`))
+        .catch(err => logger.push({ err }).log(`${logPrefix} error`));
+
+    prepareResponse(ctx);
+};
+
+const postFxTransfers = async (ctx) => {
+    const sourceFspId = ctx.request.headers['fspiop-source'];
+    const { body, headers } = ctx.request;
+    const { logger } = ctx.state;
+    const logPrefix = 'Handling POST fxTransfers request';
+
+    const model = createInboundTransfersModel(ctx);
+    model.postFxTransfers({ body, headers }, sourceFspId)
```

Comment:
You need to shallow copy the body and headers before calling the async function here. This is to avoid parallel processing issue we got previously on high concurrent calls. Please refer the functions for quotes and transfers.

Reply:
Could please, give a bit more context about issue you mentioned. For each concurrent calls we have a separate `ctx`, thus `body` and `headers` are also completely new for each concurrent calls. Sorry, but I'm not sure I get the reason, why shallow copy is needed.

Reply:
We ran into this situation with “Koa” API library we are using. This library passes a mutated context object instead of a fresh one and as a result if we pass request object by reference it will get mutated by the time async function make use of the contents of it. That's why we are shallow copying the body and headers and pass a fresh object reference. We literally got this issue when we handled high load and figured out the issue with this and this approach solved the issue. Here is a blog post we created based on our findings at that time https://www.infitx.com/async-operations-in-javascript/

Reply:
This is the old PR for this fix https://github.com/mojaloop/sdk-scheme-adapter/pull/440

Reply:
I'm not sure it's related to Koa API, coz we don't use _contex_t object in our models:
https://github.com/mojaloop/sdk-scheme-adapter/blob/feat/fx-impl-3618/modules/api-svc/src/InboundServer/handlers.js#L970
We extract needed data, and models deal only with _body, headers, sourceFspId, ..._ as with separate data (not a part of context)
Anyway, I did what you asked, please, check